src/main/kotlin/com/supergrecko/questionbot/tools/Arguments.kt

```diff
@@ -0,0 +1,13 @@
+package com.supergrecko.questionbot.tools
+
+open class Arguments(val args: List<Any?>) {
+
+    inline fun <reified T> asType(index: Int): T? {
+        return args.getOrNull(index) as? T
+    }
+
+    inline fun <reified T> fromList(index: Int, vararg indexes: Int): List<T?> {
+        return indexes.map { (args.getOrNull(index) as? List<*>)?.getOrNull(it) as? T }
```

Comment:
This makes sure the collection is the size of the vararg. This means we can successfully destructure optional arguments like ```kotlin
val (x, y) = args.fromList<Int>(0, 1, 2)
```

If the item is actually not preset, null will be returned